client/blocks/upgrade-banner/index.jsx

```diff
@@ -0,0 +1,179 @@
+/**
+ * External dependencies
+ */
+import React, { Component, PropTypes } from 'react';
+import { noop } from 'lodash';
+import classNames from 'classnames';
+
+/**
+ * Internal dependencies
+ */
+import analytics from 'lib/analytics';
+import Button from 'components/button';
+import Card from 'components/card';
+import Gridicon from 'components/gridicon';
+
+class UpgradeBanner extends Component {
+
+	static propTypes = {
+		button: PropTypes.bool,
+		callToAction: PropTypes.string,
+		callToActionButton: PropTypes.bool,
+		className: PropTypes.string,
+		color: PropTypes.string,
+		description: PropTypes.string,
+		event: PropTypes.string,
+		href: PropTypes.string,
+		icon: PropTypes.string,
+		list: PropTypes.arrayOf( PropTypes.string ),
+		onClick: PropTypes.func,
+		price: PropTypes.string,
+		title: PropTypes.string,
+	}
+
+	static defaultProps = {
+		button: false,
+		callToActionButton: false,
+		icon: 'star',
+		onClick: noop,
+	}
+
+	handleClick = () => {
+		const {
+			event,
+			href,
+			onClick,
+		} = this.props;
+
+		if ( event ) {
+			analytics.tracks.recordEvent( 'calypso_upgrade_banner_cta_click', {
+				cta_name: event,
+			} );
+		}
+
+		onClick();
+
+		if ( href ) {
+			window.location.href( href );
+		}
+	}
+
+	bannerContent() {
+		const {
+			button,
+			callToAction,
+			callToActionButton,
+			description,
+			href,
+			list,
+			title
+		} = this.props;
+
+		return (
+			<div className="upgrade-banner__content">
+				<div className="upgrade-banner__info">
+					<div className="upgrade-banner__title">
+						{ title }
+					</div>
+					<div className="upgrade-banner__description">
+						{ description }
+					</div>
+					{ list &&
```

Comment:
Considering that the purpose of this component is display some copy text, is it really something we should be worried about?
(Also, is there a reason to use `_.size` instead of `Array.length`?)